stages/org.osbuild.firewall

```diff
@@ -92,6 +119,36 @@ def main(tree, options):
                        list(map(lambda x: f"--remove-service={x}", disabled_services)),
                        check=True)
 
+    for zone_item in zones:
+        # specifying an empty zone flag results in the source being applied to
+        # the default zone
+        zone_name = zone_item['name']
+        # check that the given zone exists, if not create it
+        if zone_name != "":
+            res = subprocess.check_output(["chroot",
+                                           tree,
+                                           "firewall-offline-cmd",
+                                           "--get-zones"])
+            if res.find(zone_name.encode()) < 0:
+                subprocess.run(["chroot",
+                                tree,
+                                "firewall-offline-cmd",
+                                "--permanent",
+                                f"--new-zone={zone_name}"],
+                               check=True)
+                subprocess.run(["chroot",
+                                tree,
+                                "firewall-offline-cmd",
+                                "--reload"],
+                               check=True)
+
+        subprocess.run(["chroot",
+                        tree,
+                        "firewall-offline-cmd", f"--zone={zone_name}"] +
+                       list(map(lambda x: f"--add-source={x}",
+                                zone_item['sources'])),
+                       check=True)
```

Comment:
`sources` might be empty, right? Maybe add `minProperties=1` to the schema?

Alternatively, guard this call by `if zone_item.get("sources", []):`, this way we can add other zone-configurations in the future without having to re-arrange this call.

Reply:
Yes, `required` does indeed not prevent it from being empty. Will guard the call :+1: